src/controllers/userController.js

```diff
@@ -85,13 +109,37 @@ const updateUserPassword = async (req, res) => {
   await user.save(); // Save the updated user to the database
   // Send a JSON response with the status code 200 OK and a success message
   res.status(StatusCodes.OK).json({ msg: 'Success! Password Updated.' });
-};
 
+  /*
+     #swagger.summary = 'Fetch the current requesting user by id, update their password and save it to the database.'
+     #swagger.responses[200] = {
+				description: 'User successfully fetched and password updated.',
+				type: 'object'
+				properties: {
+				  msg: { type: 'string' }
+				}
+		 }
+		 #swagger.responses[400] = { description: 'Error. Need to provide both new and old password values' }
+     #swagger.responses[401] = { description: 'Error. Invalid credentials.' }
+  */
+};
 const deleteSingleUser = async (req, res) => {
   const userId = req.params.id;
+  if (!userId) {
+    // Throw a NotFoundError if the user is not found
+    throw new CustomError.NotFoundError(`No user with id: ${req.params.id}`);
+  }
   // Find the user by ID and delete
   await User.findByIdAndDelete(userId);
   res.status(StatusCodes.OK).json({ message: 'User deleted successfully' });
+  /*
+     #swagger.summary = 'Fetch a user by id and delete user.'
```

Comment:
Like with the getAllUsers endpoint, we can add to the description here to let frontend know that only admin users can hit this endpoint:

```suggestion
     #swagger.summary = 'Fetch a user by id and delete user.'
     #swagger.description = '**ROLE REQUIRED:** admin'
```